src/test/kotlin/graphics/scenery/tests/examples/volumes/VDIClient.kt

```diff
@@ -0,0 +1,390 @@
+import graphics.scenery.volumes.vdi.VDIDataIO
+import graphics.scenery.volumes.vdi.VDINode
+import net.imglib2.type.numeric.integer.UnsignedIntType
+import net.imglib2.type.numeric.real.FloatType
+import org.joml.Matrix4f
+import org.joml.Vector3f
+import org.joml.Vector3i
+import org.lwjgl.system.MemoryUtil
+import org.zeromq.SocketType
+import org.zeromq.ZContext
+import org.zeromq.ZMQ
+import org.zeromq.ZMQException
+import java.io.*
+import java.nio.ByteBuffer
+import java.util.concurrent.atomic.AtomicInteger
+import kotlin.concurrent.thread
+import kotlin.system.measureNanoTime
+
+class VDIClient : SceneryBase("VDI Client", 512, 512, wantREPL = false) {
+
+    var hmd: TrackedStereoGlasses? = null
+    val cam: Camera = DetachedHeadCamera(hmd)
+    val compute = VDINode()
+    val plane = FullscreenObject()
+    val context = ZContext(4)
+
+    val numSupersegments = 20
+    val skipEmpty = true
+    val vdiStreaming = true
+    var newVDI = false
+    var firstVDI = true
+
+    private val vulkanProjectionFix =
+        Matrix4f(
+            1.0f, 0.0f, 0.0f, 0.0f,
+            0.0f, -1.0f, 0.0f, 0.0f,
+            0.0f, 0.0f, 0.5f, 0.0f,
+            0.0f, 0.0f, 0.5f, 1.0f)
+
+    fun Matrix4f.applyVulkanCoordinateSystem(): Matrix4f {
+        val m = Matrix4f(vulkanProjectionFix)
+        m.mul(this)
+
+        return m
+    }
+
+    override fun init() {
+        renderer = hub.add(Renderer.createRenderer(hub, applicationName, scene, windowWidth, windowHeight))
+
+        //Step1: create Camera
+        with(cam) {
+            spatial {
+                position = Vector3f(0.0f, 0.5f, 5.0f)
+            }
+            perspectiveCamera(50.0f, windowWidth, windowHeight)
+            scene.addChild(this)
+        }
+        cam.farPlaneDistance = 20.0f
+        val opBuffer = MemoryUtil.memCalloc(windowWidth * windowHeight * 4)
+
+        //Step 2: Create vdi node and it's propertie
+        compute.name = "vdi node"
+        compute.setMaterial(ShaderMaterial(Shaders.ShadersFromFiles(arrayOf("VDIRenderer.comp"), this@VDIClient::class.java)))
+        compute.material().textures["OutputViewport"] = Texture.fromImage(Image(opBuffer, windowWidth, windowHeight),
+            usage = hashSetOf(Texture.UsageType.LoadStoreImage, Texture.UsageType.Texture))
+        compute.metadata["ComputeMetadata"] = ComputeMetadata(
+            workSizes = Vector3i(windowWidth, windowHeight, 1),
+            invocationType = InvocationType.Permanent
+        )
+        compute.visible = true
+        scene.addChild(compute)
+
+        //Step3: set plane properties
+        scene.addChild(plane)
+        plane.material().textures["diffuse"] = compute.material().textures["OutputViewport"]!!
+
+        //Step 4: call receive and update VDI
+        thread {
+            receiveAndUpdateVDI(compute)
+        }
+    }
+
+    fun updateTextures(color: ByteBuffer, depth: ByteBuffer, accelGridBuffer: ByteBuffer, vdiData: VDIData, firstVDI: Boolean) {
```

Comment:
This piece of code occurs in a few places and should likely be factored out. It might be a good idea to combine this with the code that's currently sitting the Vulkan renderer and move it into it's own class for handling VDI rendering and the required textures.